highlighters/main/main-highlighter.zsh

```diff
@@ -442,16 +509,20 @@ _zsh_highlight_highlighter_main_paint()
     # Parse the sudo command line
     if (( ! in_redirection )); then
       if [[ $this_word == *':sudo_opt:'* ]]; then
-        case "$arg" in
+        if [[ -n $flags_with_argument ]] &&
+           [[ $arg == '-'[$flags_with_argument] ]]; then
           # Flag that requires an argument
-          '-'[Cgprtu]) this_word=${this_word//:start:/};
-                       next_word=':sudo_arg:';;
+          this_word=${this_word//:start:/}
+          next_word=':sudo_arg:'
+        elif [[ $arg == '-'* ]]; then
+          # Flag that requires no argument, or unknown flag.
           # This prevents misbehavior with sudo -u -otherargument
-          '-'*)        this_word=${this_word//:start:/};
-                       next_word+=':start:';
-                       next_word+=':sudo_opt:';;
-          *)           ;;
-        esac
+          this_word=${this_word//:start:/}
+          next_word+=':start:'
```

Comment:
Might as well combine the two appends.

Reply:
That's just coding style, isn't it?

Reply:
Yes (and would be pedantically incorrect cf. next comment).